File: DotNet/Tests/Mocks/TestHydraMessage.cs
using System;
using System.Runtime.Serialization;
using Shastra.Hydra.Messaging;

namespace Shastra.Hydra.Tests.Mocks
{
    /// <summary>
    /// Extension of HydraMessage with testing-specific fields
    /// </summary>
    class TestHydraMessage : HydraMessage
    {
        /// <summary>
        /// Generate the DocId from this date instead of the current date.
        /// </summary>
        [DataMember]
        public DateTime IdDate { get; set; }
    }
}

File: Java/hydra/src/test/java/uk/co/shastra/hydra/messaging/mocks/TestHydraMessage.java
package uk.co.shastra.hydra.messaging.mocks;

import java.util.Date;

import uk.co.shastra.hydra.messaging.HydraMessage;

/**
 * Extension of HydraMessage with testing-specific fields
 *
 */
public class TestHydraMessage extends HydraMessage {

	/**
     * Generate the DocId from this date instead of the current date.
     */
    private Date idDate;
    public Date getIdDate() { return idDate; }
	public void setIdDate(Date idDate) { this.idDate = idDate; }

}
